### PR TITLE
Feature/562 protobuf bindings support

### PR DESCRIPTION
**Pull request by egaxhaj-figure**
## Description

- Publish Protobuf bindings for Kotlin and Java to Maven Central

closes: #562 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: .github/workflows/build-and-publish-proto-bindings.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Protobuf Bindings
+
+on: push
+#  pull_request:
+#  push:
+#    branches:
+#      - main
```
Comment (dwedul-figure) [inline .github/workflows/build-and-publish-proto-bindings.yml:7]: Does this need changing before merging?

Also line 35 below:
```yml
        # if: github.event_name == 'release'
```

We don't want every push to publish a new copy of this stuff.

File: .github/workflows/build-and-publish-proto-bindings.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Version
+        run: |
+          # Strip git prefix from version.
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+
+          # Strip "v" prefix from tag name.
```
Comment (minxylynx) [inline .github/workflows/build-and-publish-proto-bindings.yml:22]: Why are we stripping the v? We dont do that with the other release stuff, do we?

**Review by minxylynx — COMMENTED** [review]
So how I have things compiling, the Kotlin classes need the Java classes. Do we need to have both compile types as separate things? Just asking

File: .github/workflows/build-and-publish-proto-bindings.yml (hunk 2 above)
Comment (iramiller) [reply .github/workflows/build-and-publish-proto-bindings.yml:22]: No... the v is part of the normal go version number... we had some mixed uses before that caused issues so we try to keep the v everywhere.

File: .github/workflows/build-and-publish-proto-bindings.yml (hunk 1 above)
Comment (egaxhaj-figure) [reply .github/workflows/build-and-publish-proto-bindings.yml:7]: Yes. I will need to change this. I was using `push` to test nexus publishing. 

I'll clean up line 35. We don't need it.

File: .github/workflows/build-and-publish-proto-bindings.yml (hunk 2 above)
Comment (egaxhaj-figure) [reply .github/workflows/build-and-publish-proto-bindings.yml:22]: Sorry, I must have been working of an old example. 

Are we ok with including the `v` in the maven artifact? `io.provenance:proto-kotlin:v1.7.5`. Normally form maven it's just the version number `1.7.5`

Comment (iramiller) [reply .github/workflows/build-and-publish-proto-bindings.yml:22]: I am fine with following existing conventions if our standard doesn't make sense there

**Review by dwedul-figure — APPROVED** [review]
LGTM

**Review by iramiller — APPROVED** [review]

**Pull request merged**